### ISSUE TITLE
log method does not catch ECONNREFUSED or ECONNRESET

### ISSUE THREAD

**Issue by TheVedhaGuy** [paulgrove/node-syslog-client#23]
I have an async function that returns a promise.

When the hostname or port number is incorrect, it returns the error correctly. However, when it is a connection based issue, it does not seem to return anything thus resulting in a success.

Some quick generic code below.

      `const syslog = syslog.createClient("hostname", createOptions);

            await new Promise<void>((resolve, reject) => {
                 syslog.log("msg", logOptions, (err) => {
                     if (err) {
                         logger.error(`Failed to send to syslog server: ${err}`);
                         syslogLogger.close();
                         reject(err);
                         return;
                     } else {
                         logger.info("Sent event to syslog server...");
                         syslogLogger.close();
                         return resolve();
                     }
                 });
         });

         return;
    }`

**Comment by abizake** [comment]
@TheVedhaGuy  Any luck with this ??? For me even for wrong hosts and port it gives success.

**Comment by TheVedhaGuy** [comment]
@abizake Not really. I was using Papertrail as a cloud log provider to test syslog messages. I'm not able to confirm if it is an issue at the protocol level. With the wrong host/port I get a message back from the provider saying it is incorrect so I'm not sure why you would be facing that problem.
